.github/workflows/pre-commit.yaml

```diff
@@ -1 +1,23 @@
 name: pre-commit
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  style:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+      - name: Check out repos
+        uses: actions/checkout@v2
+      - name: Install lint utilities
+        run: |
+          pip install pre-commit
+          pre-commit install-hooks
+      - name: Run precommit hooks for linting
+        run: pre-commit run --all-files
```

Comment:
As a routine check, do we need to run pre-commit on all files? should we just run changes only?

Reply:
this is only trial pr to see if workflow runs. will probably replace precommit with actual lints

Reply:
hm it might be noisy though, if some unrelated changes get merged, as this is not diff blocking yet. just like it failed for me. 
so the one used by torchrec is apparently deprecated as per readme so didnt go with that

Reply:
Q: If the pre-commit CI fails, will it give the developer a warning page or auto format his/her code right away?

Sorry didn't notice the DEPRECATED notice.

Reply:
updated the summary. have a followup